2017-04-30  Arpit Deepak Jasapara  <arpit@lnxsrv07.seas.ucla.edu>

	* timespec.h (timespec_cmp): Used comparison operators to prevent overflow issues.

--- coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec.h	2017-04-30 23:24:32.188226000 -0700
@@ -45,8 +45,11 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec > b.tv_sec)
+    return 1;
+  if (a.tv_sec < b.tv_sec)
+    return -1;
+  return a.tv_nsec - b.tv_nsec;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
